Tests/MongoSwiftSyncTests/MongoCollectionTests.swift

```diff
@@ -256,6 +256,31 @@ final class MongoCollectionTests: MongoSwiftTestCase {
         expect(try self.coll.deleteOne(["cat": "cat"])?.deletedCount).to(equal(1))
```

Comment:
can we amend the `testHintInPreviousVersion` test to include `findOneAndDelete`? From looking at the spec, it seems it is only required for `findAndModify` based commands and isn't required for the other ones though.

As a side note: can we update that test name to include a reference to `findAndModify` to clarify that?